Packs/Rapid7_Nexpose/Integrations/Rapid7_Nexpose/Rapid7_Nexpose.py

```diff
@@ -5376,6 +5582,442 @@ def update_vulnerability_exception_status_command(client: Client, vulnerability_
+    """
+    site_id_int = arg_to_number(site_id, arg_name="site_id", required=True)
+
+    if asset_type == "assets":
+        res = client.send_http_request("GET", f"/sites/{site_id_int}/{target_type}_targets")
+        output_prefix = f"Nexpose.{target_type.capitalize()}Asset"
+        readable_title = f"{target_type.capitalize()} Asset list for site ID {site_id_int}"
+    elif asset_type == "asset_groups":
+        res = client.send_http_request("GET", f"/sites/{site_id_int}/{target_type}_asset_groups")
+        output_prefix = f"Nexpose.{target_type.capitalize()}AssetGroup"
+        readable_title = f"{target_type.capitalize()} Asset group list for site ID {site_id_int}"
+    else:
+        raise ValueError("Invalid asset_type. Expected 'assets' or 'asset_groups'.")
+
+    outputs = dict(**res, site_id=site_id_int)
+    res = remove_empty_elements(res)
+
+    return CommandResults(
+        outputs_prefix=output_prefix,
+        outputs_key_field="id",
+        outputs=outputs,
+        readable_output=tableToMarkdown(readable_title, res, headerTransform=string_to_table_header),
+        raw_response=outputs
+    )
+
+
+def create_asset_group_command(client: Client, name: str, description: str, type: str, match: str | None = None,
+                               ip_address_is: str | None = None, host_name_is: str | None = None,
+                               risk_score_higher_than: str | None = None, vulnerability_title_contains: str | None = None,
+                               site_id_in: str | None = None, site_name_in: str | None = None, query: str | None = None, ):
+    filters_data = parse_asset_filters(
+        client=client,
+        ip_address_is=ip_address_is,
+        host_name_is=host_name_is,
+        risk_score_higher_than=risk_score_higher_than,
+        site_id_in=site_id_in,
+        site_name_in=site_name_in,
+        vulnerability_title_contains=vulnerability_title_contains,
+        query=query)
+
+    if type == "dynamic" and not filters_data:
+        raise DemistoException("you must use filters to create a dynamic asset group")
+
+    filters = convert_asset_search_filters(filters_data)
+
+    res = client.create_asset_group(name=name, description=description, type=type, filters=filters, match=match)
+
+    return CommandResults(
+        outputs_prefix="Nexpose.AssetGroup",
+        outputs_key_field="id",
+        outputs=res,
+        readable_output=f"A new asset group {name} created successfully with ID: {res.get('id', 'ID not found')}",
+        raw_response=res
+    )
+
+
+def get_list_asset_group_command(client: Client, group_id: str | None = None, group_name: str | None = None,
+                                 type: str | None = None, page_size: str | None = None, page: str | None = None,
+                                 limit: str | None = None, sort: str | None = None):
+    if id_int := arg_to_number(group_id, required=False):
+        asset_groups = client.get_asset_group_by_id(id=id_int)
```

Comment:
You can combine both of the request to 1 function.

Reply:
The get_asset_groups function uses pagination, whereas the get_asset_group_by_id function does not. Therefore  i prefer not to unless you say so